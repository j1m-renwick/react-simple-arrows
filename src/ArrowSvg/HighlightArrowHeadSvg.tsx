import React from 'react';

export const HighlightArrowHeadSvg = ({ id, length, width, color = 'black' }: { id: string; length: number; width: number; color?: string }) => {
  const arrowPath = `M0,0 L0,${width} L${width},${length} L${length},0 z`;
  return (
      // setting markerUnits="userSpaceOnUse" so that it doesn't inherit stroke width from the highlight path
    <marker id={id} key={id} markerWidth={length} markerHeight={width} refX={length - width / 2} refY={width / 2} orient="auto" markerUnits="userSpaceOnUse">
      <path d={arrowPath} fill={color} />
    </marker>
  );
};
