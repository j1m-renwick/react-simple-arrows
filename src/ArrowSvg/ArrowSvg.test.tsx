import React from 'react';

import { render } from '@testing-library/react';

import { LineOrientation } from '../constants';
import { ArrowSvg } from './ArrowSvg';

jest.mock('uuid', () => () => '__UUID__');

describe('ArrowSvg', () => {

  it('should look correct going down and right', async () => {
    const { container } = render(<ArrowSvg start={{ x: 0, y: 0 }} end={{ x: 500, y: 300 }} orientation={LineOrientation.HORIZONTAL} />);
    expect(container.firstChild).not.toEqual(null);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should look correct going down and left', async () => {
    const { container } = render(<ArrowSvg start={{ x: 500, y: 0 }} end={{ x: 0, y: 300 }} orientation={LineOrientation.HORIZONTAL} />);
    expect(container.firstChild).not.toEqual(null);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should look correct going down and right with custom width and color', async () => {
    const { container } = render(
        <ArrowSvg start={{ x: 0, y: 0 }} end={{ x: 500, y: 300 }} orientation={LineOrientation.HORIZONTAL} width="3" color="green" />);
    expect(container.firstChild).not.toEqual(null);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should look correct going down and left with custom width and color', async () => {
    const { container } = render(
        <ArrowSvg start={{ x: 500, y: 0 }} end={{ x: 0, y: 300 }} orientation={LineOrientation.HORIZONTAL} width="3" color="green" />);
    expect(container.firstChild).not.toEqual(null);
    expect(container.firstChild).toMatchSnapshot();
  });
});
