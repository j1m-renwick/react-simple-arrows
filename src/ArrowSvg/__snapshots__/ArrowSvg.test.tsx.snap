// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowSvg should look correct going down and left 1`] = `
<svg
  height="309"
  style="position: absolute; top: -4.5px; left: -4.5px;"
  width="509"
>
  <defs>
    <marker
      id="__UUID__"
      markerHeight="9"
      markerUnits="userSpaceOnUse"
      markerWidth="15"
      orient="auto"
      refX="10.5"
      refY="4.5"
    >
      <path
        d="M0,0 L0,9 L14,4.5 z"
        fill="black"
      />
    </marker>
  </defs>
  <path
    d="M 504.5 4.5 C 204.5 4.5 304.5 304.5 4.5 304.5"
    fill="none"
    marker-end="url(#__UUID__)"
    stroke="black"
    stroke-width="1"
  />
</svg>
`;

exports[`ArrowSvg should look correct going down and left with custom width and color 1`] = `
<svg
  height="309"
  style="position: absolute; top: -4.5px; left: -4.5px;"
  width="509"
>
  <defs>
    <marker
      id="__UUID__"
      markerHeight="9"
      markerUnits="userSpaceOnUse"
      markerWidth="15"
      orient="auto"
      refX="10.5"
      refY="4.5"
    >
      <path
        d="M0,0 L0,9 L14,4.5 z"
        fill="green"
      />
    </marker>
  </defs>
  <path
    d="M 504.5 4.5 C 204.5 4.5 304.5 304.5 4.5 304.5"
    fill="none"
    marker-end="url(#__UUID__)"
    stroke="green"
    stroke-width="3"
  />
</svg>
`;

exports[`ArrowSvg should look correct going down and right 1`] = `
<svg
  height="309"
  style="position: absolute; top: -4.5px; left: -4.5px;"
  width="509"
>
  <defs>
    <marker
      id="__UUID__"
      markerHeight="9"
      markerUnits="userSpaceOnUse"
      markerWidth="15"
      orient="auto"
      refX="10.5"
      refY="4.5"
    >
      <path
        d="M0,0 L0,9 L14,4.5 z"
        fill="black"
      />
    </marker>
  </defs>
  <path
    d="M 4.5 4.5 C 304.5 4.5 204.5 304.5 504.5 304.5"
    fill="none"
    marker-end="url(#__UUID__)"
    stroke="black"
    stroke-width="1"
  />
</svg>
`;

exports[`ArrowSvg should look correct going down and right with custom width and color 1`] = `
<svg
  height="309"
  style="position: absolute; top: -4.5px; left: -4.5px;"
  width="509"
>
  <defs>
    <marker
      id="__UUID__"
      markerHeight="9"
      markerUnits="userSpaceOnUse"
      markerWidth="15"
      orient="auto"
      refX="10.5"
      refY="4.5"
    >
      <path
        d="M0,0 L0,9 L14,4.5 z"
        fill="green"
      />
    </marker>
  </defs>
  <path
    d="M 4.5 4.5 C 304.5 4.5 204.5 304.5 504.5 304.5"
    fill="none"
    marker-end="url(#__UUID__)"
    stroke="green"
    stroke-width="3"
  />
</svg>
`;
